vassal-app/src/main/java/VASSAL/counters/NonRectangular.java

```diff
@@ -58,9 +58,11 @@
 public class NonRectangular extends Decorator implements EditablePiece {
   public static final String ID = "nonRect;"; // NON-NLS
   private static final Map<String, Shape> shapeCache = new HashMap<>();
+  private static final Map<String, String> nameCache = new HashMap<>();
```

Comment:
Rather than having two parallel maps for this, I would use one map where the value is either a two-member data class with members `name` and `shape`, or an `org.apache.commons.lang3.Pair<String, Shape>`.

Reply:
Updated.